dpctl/_sycl_device.pyx

```diff
@@ -157,6 +179,72 @@ cdef class SyclDevice(_SyclDevice):
             SyclDevice._init_helper(self, DRef)
             return 0
 
+    cpdef list create_sub_devices_equally(self, size_t count):
+        """ Returns a vector of sub devices partitioned from this SYCL device
+            based on the count parameter. The returned
+            vector contains as many sub devices as can be created such that each sub
+            device contains count compute units. If the device’s total number of compute
+            units is not evenly divided by count, then the remaining compute units are
+            not included in any of the sub devices.
+        """
+        cdef DPCTLDeviceVectorRef DVRef = NULL
+        DVRef = DPCTLDevice_CreateSubDevicesEqually(self._device_ref, count)
+        cdef list devices = _get_devices(DVRef)
+        DPCTLDeviceVector_Delete(DVRef)
+        return devices
+
+    cpdef list create_sub_devices_by_counts(self, size_t[:] counts, size_t ncounts):
+        """ Returns a vector of sub devices
+            partitioned from this SYCL device based on the counts parameter. For each
+            non-zero value M in the counts vector, a sub device with M compute units
+            is created.
+        """
+        cdef size_t *counts_buff = &counts[0]
+        cdef DPCTLDeviceVectorRef DVRef = NULL
+        DVRef = DPCTLDevice_CreateSubDevicesByCounts(self._device_ref, counts_buff, ncounts)
+        cdef list devices = _get_devices(DVRef)
+        DPCTLDeviceVector_Delete(DVRef)
+        return devices
+
+    cpdef list create_sub_devices_by_affinity(self, _partition_affinity_domain_type domain):
```

Comment:
As @oleksandr-pavlyk  said, these functions need not be exposed to Python and can be `cdef`.